### PR TITLE
Finalise integration tests

### PR DESCRIPTION
**Pull request by cargopete**
- Make tests more robust and fix DB clearing earlier than it should.
- Add test for count of DB entries.
- Change mock indexers with working ones (taken from testnet).

To get the best log format:
`yarn integration-tests | pino-pretty -c -f`

### CODE REVIEW

File: src/examples/poi-crosschecker/tests/integration/compareAttestations.ts [hunk 1]
```diff
@@ -1,58 +1,78 @@
...
-    async: false,
-    // eslint-disable-next-line @typescript-eslint/no-explicit-any
-    level: process.env.logLevel as any,
-  });
-
-  const db = new sqlite3.Database(
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export const compareAttestations = async (logger: Logger) => {
+  const db = openDb(
     "/usr/app/dist/src/examples/poi-crosschecker/npois.db",
```
Comment (hopeyen) [inline src/examples/poi-crosschecker/tests/integration/compareAttestations.ts]: maybe for consistency we should update the name from `npois.db` to `poi_crosschecker.db`?

**Review by hopeyen — COMMENTED** [review]
the tests look great! It covers a lot of types of tests we can do.
I think all we need is rebase the PR to main and squash some commits - I had changed `npois.db` to `poi_crosschecker.db` so I think some of the db open should be updated as well:)

File: src/examples/poi-crosschecker/tests/integration/compareAttestations.ts (hunk 1 above)
Comment (cargopete) [reply src/examples/poi-crosschecker/tests/integration/compareAttestations.ts]: Ah yep, missed that one

**Comment by axiomatic-aardvark** [comment]
@hopeyen  do you think it's ok to have the **database** name `poi_crosschecker` but keep the **table** name `npois` ?

**Review by hopeyen — APPROVED** [review]
🔥 great work!

**Pull request merged**